File: 2_Kernel/JavaKernel/src/main/java/com/seagosoft/tws/data/RealTimeBarData.java
package com.seagosoft.tws.data;

import com.ib.client.Decimal;

public class RealTimeBarData {

    public int reqId;
    public long time;
    public double open;
    public double high;
    public double low;
    public double close;
    public Decimal volume;
    public Decimal wap;
    public int count;

    public RealTimeBarData(int reqId, long time, double open, double high, double low,
                           double close, Decimal volume, Decimal wap, int count) {
        this.reqId = reqId;
        this.time = time;
        this.open = open;
        this.high = high;
        this.low = low;
        this.close = close;
        this.volume = volume;
        this.wap = wap;
        this.count = count;
    }
}

File: 3_OrderSystem/KotlinEWrapperBackend/src/main/kotlin/com/seagosoft/tws/wrapper/structs/RealTimeBarData.kt
package com.seagosoft.tws.wrapper.structs

import com.ib.client.Decimal


class RealTimeBarData(
    var reqId: Int, var time: Long, var open: Double, var high: Double, var low: Double,
    var close: Double, var volume: Decimal, var wap: Decimal, var count: Int
)